sound/soc/sof/ipc4-pcm.c

```diff
@@ -495,17 +496,32 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 				use_chain_dma = true;
 		}
 	}
+
+	fe_rate = params_rate(params);
+
 	/*
-	 * Always set BE format to 32-bits for both playback and capture, except if chained
-	 * DMA is in use. In chained DMA case the back-end and front-end format must match.
+	 * Limit sampling rate if copier has a format that matches
+	 * the rate.
 	 */
 	if (!use_chain_dma) {
-		snd_mask_none(fmt);
-		snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S32_LE);
+		struct sof_ipc4_available_audio_format *available_fmt = &ipc4_copier->available_fmt;
+		int i;
+
+		for (i = 0; i < available_fmt->audio_fmt_num; i++)
```

Comment:
@kv2019i do you need to rebase your PR to take in my changes? We no longer have audio_fmt_num

Reply:
Ack, rebased!